src/components/molecules/TableComponent/TableComponent.tsx

```diff
@@ -15,55 +33,149 @@ const TableComponent: React.FunctionComponent<TableComponentPropsType> = ({
   table,
   tableLocked,
   venue,
+  type,
 }) => {
   const { openUserProfileModal } = useProfileModalControls();
   const locked = tableLocked(table.reference);
   const numberOfSeatsLeft = table.capacity && table.capacity - users.length;
   const full = numberOfSeatsLeft === 0;
-  return (
-    <div className={`table-component-container ${table.reference}`}>
-      <div
-        className="table-item"
-        style={{
+  const { userId } = useUser();
+
+  const { isAdminUser, isLoading: isCheckingRole } = useIsAdminUser(userId);
+
+  const isRemoveButtonShown = !isCheckingRole && isAdminUser;
+
+  const {
+    isShown: isModalShown,
+    toggle: toggleModal,
+    hide: hideModal,
+  } = useShowHide(false);
+
+  const [{ loading: isDeletingTable }, removeTable] = useAsyncFn(async () => {
+    if (!venue.id) return;
+
+    await deleteTable({
+      venueId: venue.id,
+      tableName: table.reference,
+      defaultTables:
+        venue.template === VenueTemplate.jazzbar
+          ? JAZZBAR_TABLES
+          : CONVERSATION_TABLES,
+    });
+
+    toggleModal();
+  }, [table.reference, venue.id, venue.template, toggleModal]);
+
+  const isDeleteButtonDisabled = isDeletingTable;
+
+  const isJazzBar = type === VenueTemplate.jazzbar;
+
+  const itemStyles = useCss(
+    isJazzBar
+      ? {}
+      : {
           height: `${table.rows && table.rows * 50 + 65}px`,
           width: `${table.columns && (table.columns + 1) * 55}px`,
-        }}
-      >
-        <div className="table-occupancy-information red-text">
+        }
+  );
+
+  const itemClasses = classNames("TableComponent__item", itemStyles);
+
+  const itemWrapperStyles = useCss({
+    width: isJazzBar ? "60%" : "65%",
+  });
+
+  const itemWrapperClasses = classNames(
+    "TableComponent__item-wrapper",
+    itemWrapperStyles
+  );
+
+  return (
+    <div className={`TableComponent ${table.reference}`}>
```

Comment:
Is it possible to prefix this?

```suggestion
    <div className={`TableComponent TableComponent--ref-${table.reference}`}>
```

Reply:
removed this reference as its unnecessary 2ba4481